src/frontend/mame/ui/state.cpp

```diff
@@ -0,0 +1,424 @@
+// license:BSD-3-Clause
+// copyright-holders:Nathan Woods
+/***************************************************************************
+
+    ui/state.cpp
+
+    Menus for saving and loading state
+
+***************************************************************************/
+
+#include "emu.h"
+#include "machine.h"
+#include "emuopts.h"
+#include "ui/state.h"
```

Comment:
Why explicitly `#include "machine.h"` here?  It's very much part of `emu.h`.  Also, unit header (in this case `ui/state.h`) should be right after emu.h to help avoid weird issues where headers require you to `#include` other headers before them.